specs/FileTypePolicy.md

```diff
@@ -0,0 +1,222 @@
+            {
+                // Get the file extension for file to be saved.
+                // And create your own rules of file type policy.
+                LPWSTR extension;
+                CHECK_FAILURE(args->get_FileExtension(&extension));
+                if (lstrcmpW(extension, L"eml") == 0)
+                {
+                    // Set the `SuppressDefaultPolicy` property to skip the
+                    // default file type policy checking and a possible security
+                    // alert dialog for "eml" file. This will consent to
+                    // save the file.
+                    CHECK_FAILURE(args->put_SuppressDefaultPolicy(TRUE));
+                }
+                if (lstrcmpW(extension, L"exe") == 0)
+                {
+                    // Set the `Cancel` property to cancel the saving for "exe"
+                    // file directly. Save action will be aborted without any
+                    // alert.
+                    CHECK_FAILURE(args->put_Cancel(TRUE));
+                }
+                return S_OK;
+            })
+            .Get(),
+        &m_saveFileSecurityCheckStartingToken);
+    return true;
+}
+```
+
+## .Net/ WinRT
+This example shows suppressing file type policy, security dialog, and 
+allow to save the file directly. It also blocks saving the exe file.
+The sample code will register the event with custom rules.
+```c#
+void AddCustomFileTypePoliciesExecuted(object target, ExecutedRoutedEventArgs e)
+{
+    // Register a handler for the `SaveFileSecurityCheckStarting` event.
+    webView.CoreWebView2.SaveFileSecurityCheckStarting += (sender, args) =>
+    {
+        // Get the file extension for file to be saved.
+        // And create your own rules of file type policy.
+        if (args.FileExtension == "eml")
+        {
+            // Set the `SuppressDefaultPolicy` property to skip the
+            // default file type policy checking and a possible security
+            // alert dialog for "eml" file. This will consent to
+            // save the file.
+            args.SuppressDefaultPolicy = true;
+        }
+        if (args.FileExtension == "exe")
+        {
+            // Set the `Cancel` property to cancel the saving for "exe"
+            // file directly. Save action will be aborted without any
+            // alert.
+            args.Cancel = true;
+        }
+    };
+}
+```
+
+# API Details
+## Win32 C++
+```c++
+interface ICoreWebView2 : IUnknown {
+  /// Adds an event handler for the `SaveFileSecurityCheckStarting` event.
+  /// If the default save file picker is used to save a file, for
+  /// example, client using the File System API `showSaveFilePicker`;
+  /// this event will be raised during system FileTypePolicy
+  /// checking the dangerous file extension list.
+  /// 
+  /// Developers can specify their own decision on if allow this file 
+  /// type extension to be saved, or downloaded. Here are two properties
```

Comment:
>Developers can specify their own decision on if allow this file type extension to be saved, or downloaded.

A little unclear. Maybe something like:
"Developers can specify their own decision on whether to allow this file type extension to be saved, or downloaded."

Reply:
Also, the developer can make a decision based on criteria other than just file extension. So maybe the comment should be along the lines of:
"Developers can specify their own logic for determining whether to allow a particular file to be downloaded, based on factors such as the file extension or the origin Uri".

Reply:
Please fix. Note that the criteria doesn't have to be based solely on the file extension.